lib/galaxy/jobs/__init__.py

```diff
@@ -786,6 +790,10 @@ def get_parallelism(self):
         return self.tool.parallelism
 
     @property
+    def shell(self):
+        return self.job_destination.shell or getattr(self.app.config, 'default_job_shell', DEFAULT_JOB_SHELL)
```

Comment:
I think that if they are not equivalent in this case, then that should be fixed. Isn't it the whole point of an interface to hide where the property is actually stored?